core/src/main/java/org/kiwix/kiwixmobile/core/ViewModelFactory.kt

```diff
@@ -28,7 +28,7 @@ open class ViewModelFactory @Inject constructor(
 ) : ViewModelProvider.Factory {
 
   @Suppress("TooGenericExceptionCaught", "TooGenericExceptionThrown")
-  override fun <T : ViewModel?> create(modelClass: Class<T>): T {
+  override fun <T : ViewModel> create(modelClass: Class<T>): T {
```

Comment:
Hi @gouri-panda ,
We can not make it nullable after upgrading project. It gives compile time error. Because in older version we have 
```
@NonNull
<T extends ViewModel> T create(@NonNull Class<T> modelClass);
``` 
and it gives us warning to make it NonNull. But now we have 
``` 
public fun <T : ViewModel> create(modelClass: Class<T>, extras: CreationExtras): T =
            create(modelClass)
```

```
public fun <T : ViewModel> create(modelClass: Class<T>): T {
           throw UnsupportedOperationException(
                "Factory.create(String) is unsupported.  This Factory requires " +
                    "`CreationExtras` to be passed into `create` method."
            )
        }
```

and we can not  make it nullable. So have to make it NonNull